pkg/readiness/ready_tracker.go

```diff
@@ -484,6 +510,31 @@ func (t *Tracker) trackModifySet(ctx context.Context) error {
 	return nil
 }
 
+func (t *Tracker) trackExternalDataProvider(ctx context.Context) error {
+	defer func() {
+		t.externalDataProvider.ExpectationsDone()
+		log.V(1).Info("Provider expectations populated")
+		_ = t.constraintTrackers.Wait()
+	}()
+
+	if !t.mutationEnabled {
```

Comment:
`mutationEnabled` is probably the wrong var to check here.

Reply:
updated, thanks!